vmd-api/api/openapi.yaml

```diff
@@ -0,0 +1,325 @@
+        default:
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorModel'
+          description: error payload
+      summary: Request to perform a control action on the virtual machine by its ID
+      tags:
+      - vm
+components:
+  schemas:
+    ErrorModel:
+      properties:
+        code:
+          example: 0
+        message:
+          example: SomeError#0
+      required:
+      - code
+      - message
+    VirtualMachineInfo:
+      example:
+        memory:
+          maximumAllocation: 2048
+          currentAllocation: 2048
+          totalHost: 8192
+        performance:
+          cpuUsage: 50
+          diskIo: ""
+          memoryUsage:
+            numerator: ""
+            denominator: ""
+          networkIo: ""
+        os:
+          name: Ubuntu
+          version: 23.04
+        devices:
+          list: ""
+        hypervisorDetails:
+          emulator: /usr/bin/qemu-system-x86_64
+          hypervisor: KVM
+          arch: aarch64
+          chipset: Q35
+          firmware: BIOS
+        cpu:
+          cores: 4
+          threads: 8
+          model: NVidia Tegra X1
+          sockets: 1
+          logicalCount: 8
+        basicDetails:
+          name: GUI VM
+          description: GUI Virtual Machine
+          id: 101
+          state: R
+        startTime: ""
+        runningTime: ""
+      properties:
+        basicDetails:
+          $ref: '#/components/schemas/VirtualMachineBasicDetails'
+        hypervisorDetails:
+          $ref: '#/components/schemas/VirtualMachineHypervisorDetails'
+        cpu:
+          $ref: '#/components/schemas/VirtualMachineCpu'
+        memory:
+          $ref: '#/components/schemas/VirtualMachineMemory'
+        startTime: {}
+        runningTime: {}
+        os:
+          $ref: '#/components/schemas/VirtualMachineOs'
+        performance:
+          $ref: '#/components/schemas/VirtualMachinePerformance'
+        devices:
+          $ref: '#/components/schemas/VirtualMachineDevices'
+      required:
+      - basicDetails
+      - cpu
+      - memory
+      - os
+    VirtualMachineBasicDetails:
+      example:
+        name: GUI VM
+        description: GUI Virtual Machine
+        id: 101
+        state: R
+      properties:
+        id:
+          example: 101
+        state:
+          enum:
+          - D
+          - I
+          - R
+          - S
+          - T
+          - TDebug
+          - W
+          - X
+          - Z
+          example: R
+        name:
+          example: GUI VM
+        description:
+          example: GUI Virtual Machine
+      required:
+      - id
+      - name
+      - state
+    VirtualMachineHypervisorDetails:
+      example:
+        emulator: /usr/bin/qemu-system-x86_64
+        hypervisor: KVM
+        arch: aarch64
+        chipset: Q35
+        firmware: BIOS
+      properties:
+        hypervisor:
+          enum:
+          - KVM
+          - Xen
```

Comment:
We won't support `Xen`. It's misleading and should be removed but is not a blocker.

Reply:
I will note this and move it to the next PR

Reply:
Resolved in https://github.com/tiiuae/vmd/pull/35/commits/751bf9b99f2cff749e816f0ae279fd2167af8915